### PR TITLE
Java auto config: add  suppressing instr section

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to #1229
- This adds [docs/suppressing-instrumentation.md] to this repo, as a subsection of Agent config.
- I've chosen to migrate the page as is. I'll make a revisions in a followup PR so that it will be easier to track changes.

Preview: https://deploy-preview-1237--opentelemetry.netlify.app/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-auto-instrumentation

[docs/suppressing-instrumentation.md]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3ab1b8516feb5a0065d5b1e0d6a242049f2c7aea/docs/suppressing-instrumentation.md

### CODE REVIEW

File: content/en/docs/instrumentation/java/automatic/agent-config.md [hunk 1]
```diff
@@ -69,8 +69,189 @@ See [common instrumentation configuration properties][common-config].
...
+partially applied, which could have unknown side-effects.
+
+If you find yourself needing to use this, it would be great if you could drop us
+an issue explaining why, so that we can try to come up with a better solution to
+address your need.
+
+| System property                | Environment variable           | Purpose                                                                                           |
+|--------------------------------|--------------------------------|---------------------------------------------------------------------------------------------------|
+| otel.javaagent.exclude-classes | OTEL_JAVAAGENT_EXCLUDE_CLASSES | Suppresses all instrumentation for specific classes, format is "my.package.MyClass,my.package2.*" |
+
```
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/agent-config.md:211]: @open-telemetry/java-instrumentation-approvers do you know of anyone using this? it's a pretty advanced and dangerous setting, so my initial though is to remove it from this page and find another less prominent location for it

**Review by trask — APPROVED** [review]
I'm good merging as is and following up on my comment below separately

File: content/en/docs/instrumentation/java/automatic/agent-config.md (hunk 1 above)
Comment (mateuszrzeszutek) [reply content/en/docs/instrumentation/java/automatic/agent-config.md:211]: > my initial though is to remove it from this page and find another less prominent location for it

👍 

I remember we had a couple issues when people actually tried using it and it didn't work as expected. Maybe we can keep this one in the instrumentation repo? In some advanced/internal config options doc.

File: content/en/docs/instrumentation/java/automatic/agent-config.md [hunk 2]
```diff
@@ -69,8 +69,189 @@ See [common instrumentation configuration properties][common-config].
...
+### Enable manual instrumentation only
+
+You can suppress all auto instrumentations but have support for manual
+instrumentation with `@WithSpan` and normal API interactions by using
+`-Dotel.instrumentation.common.default-enabled=false
+-Dotel.instrumentation.opentelemetry-annotations.enabled=true`
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/automatic/agent-config.md:217]: This should enable `opentelemetry-api` too:
```suggestion
`-Dotel.instrumentation.common.default-enabled=false
-Dotel.instrumentation.opentelemetry-api.enabled=true
-Dotel.instrumentation.opentelemetry-annotations.enabled=true`
```

File: content/en/docs/instrumentation/java/automatic/agent-config.md (hunk 1 above)
Comment (trask) [reply content/en/docs/instrumentation/java/automatic/agent-config.md:211]: thx @mateuszrzeszutek. @chalin let's go ahead and remove this section, I will find another location for it in the instrumentation repo

Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/agent-config.md]: Section removed.

**Pull request merged**